### ISSUE TITLE
Slow start of flatpak application with Dash-to-dock enabled

### ISSUE THREAD

**Issue by emagra** [micheleg/dash-to-dock#2278]
When I run the Flatpak version of Chromium and Edge (which I use for work, sob), they take several minutes to start. With Dash to Dock disabled, I have no issues. The slow startup occurs only with Chromium and Edge; other apps start without delay. Additionally, if I start them with the commands `gtk-launch org.chromium.Chromium` or `flatpak run org.chromium.Chromium`, there is no delay

OS: Debian GNU/Linux trixie/sid x86_64 
Kernel: 6.10.6-amd64 
DE: GNOME 46.4 
WM: Mutter 
Dash-to-dock: V95

**Comment by 3v1n0** [comment]
The problem happens also when running the app from the overview or just from the dock?

**Comment by vanvugt** [comment]
See also #2266

**Comment by emagra** [comment]
> The problem happens also when running the app from the overview or just from the dock?

both

**Comment by karuboniru** [comment]
I have the same iss, with dash-to-dock enabled in Fedora 41 with GNOME 47. After login, the starting of first application takes ~20+ seconds to appear its window. This includes both flatpak apps and non-flatpak apps. And when the first application started no further problem is present. 

After checking the log, it seems that the issue is connected to the time comsumed to start `xdg-desktop-portal-gnome.service` and/or `xdg-desktop-portal.service`. (I don't know which one is the reason which is the consequence though) .

With dash to dock enabled, those 2 systemd user service would take ~20s to start compared ~1s in normal case. Hope this can help addressing the cause of the issue.

**Comment by 3v1n0** [comment]
It could be related to https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/merge_requests/176 ?

**Comment by emagra** [comment]
> It could be related to https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/merge_requests/176 ?

That's not the case.

The problem is present only for Flatpak apps in particular for web browsers, in my case Chromoium and Edge. All other deb apps run with no delay even after the login.

**Comment by marvin-te** [comment]
Not much going on in this thread for a while, but I do have the same problem. Running VanillaOS Orchid (based on Debian Sid) and Gnome 46.

When DashToDock is enabled, launching Firefox from either the Dock or the Grid takes ages (minutes), launching it via `flatpak run` is near instant (I guess as instant as flatpaks get). 
Disabling DashToDock resolves the issue.

Are there any updates to this bug?

**Comment by marvin-te** [comment]
Hi @3v1n0, thanks for checking in. Sadly this does not solve the issue :/

**Comment by 3v1n0** [comment]
Ok I'm quite happy it doesn't 😅, but I guess we've to go through trying disabling various elements to understand how to make things work properly...

Per se we don't touch the overview much... Another thing that could interact is the unity notifications support, try to toggle that instead

**Comment by marvin-te** [comment]
It's the `Isolate Workspace` and `Isolate Monitors` options. Just fiddled around with about each and every toggle in the "Launchers" section. Those seem to be the only ones effecting this issue. Is either of them on, Firefox takes ages to start. For reproduction: A shell restart is indeed necessary after each settings change.

**Comment by emagra** [comment]
> It's the `Isolate Workspace` and `Isolate Monitors` options. Just fiddled around with about each and every toggle in the "Launchers" section. Those seem to be the only ones effecting this issue. Is either of them on, Firefox takes ages to start. For reproduction: A shell restart is indeed necessary after each settings change.

I can confirm. With `Isolate Workspace` and `Isolate Monitors` options disabled the app start instantaneously.

**Comment by Tuquilo9** [comment]
I have the same issue but I haven't found any fixes if anyone has one I would like to get it ! Thanks!

**Comment by VednaGames** [comment]
> It's the `Isolate Workspace` and `Isolate Monitors` options. Just fiddled around with about each and every toggle in the "Launchers" section. Those seem to be the only ones effecting this issue. Is either of them on, Firefox takes ages to start. For reproduction: A shell restart is indeed necessary after each settings change.

When disabling those two functions application launch does speed up, but if the app is launched from an option in the right click menu, for example "new window", the problem persists and the app opens slowly.

**Comment by jlehtoranta** [comment]
This is most likely not a dash-to-dock related issue:
* https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/7958
* https://github.com/flatpak/flatpak/issues/6022

Please confirm either by installing [flatpak 1.15.x pre-release](https://github.com/flatpak/flatpak/issues/5968#issuecomment-2494528039) or by [enabling an apparmor profile](https://github.com/flatpak/flatpak/issues/6022#issuecomment-2552229473).

**Comment by emagra** [comment]
> This is most likely not a dash-to-dock related issue:
> 
>     * https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/7958
> 
>     * [[Bug]: 100% CPU when launching an app with a large RLIMIT_NOFILE soft limit flatpak/flatpak#6022](https://github.com/flatpak/flatpak/issues/6022)
> 
> 
> Please confirm either by installing [flatpak 1.15.x pre-release](https://github.com/flatpak/flatpak/issues/5968#issuecomment-2494528039) or by [enabling an apparmor profile](https://github.com/flatpak/flatpak/issues/6022#issuecomment-2552229473).

Yes, I can confirm by [enabling an apparmor profile](https://github.com/flatpak/flatpak/issues/6022#issuecomment-2552229473). 

But with some changes in the `/etc/apparmor.d/flatpak` file, because I have AppArmor 3.1.7

```
# This profile allows everything and only exists to give the
# application a name instead of having the label "unconfined"

abi <abi/3.0>,

include <tunables/global>

profile flatpak /usr/bin/flatpak flags=(attach_disconnected,mediate_deleted) {
  # userns,

  # Site-specific additions and overrides. See local/README for details.
  include if exists <local/flatpak>
}

```

commend userns beacause generates a syntax error, unexpected TOK_END_OF_RULE, expecting TOK_MODE. But I don't know what I'm doing, I'm completely new to apparmor.

**Comment by VednaGames** [comment]
> This is most likely not a dash-to-dock related issue:
> 
>     * https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/7958
> 
>     * [[Bug]: 100% CPU when launching an app with a large RLIMIT_NOFILE soft limit flatpak/flatpak#6022](https://github.com/flatpak/flatpak/issues/6022)
> 
> 
> Please confirm either by installing [flatpak 1.15.x pre-release](https://github.com/flatpak/flatpak/issues/5968#issuecomment-2494528039) or by [enabling an apparmor profile](https://github.com/flatpak/flatpak/issues/6022#issuecomment-2552229473).

I can confirm by installing [flatpak 1.15.x pre-release](https://github.com/flatpak/flatpak/issues/5968#issuecomment-2494528039) that this issue doesn't happen anymore.
